packages/@aws-cdk/core/lib/cfn-json-provider/index.ts

```diff
@@ -0,0 +1,10 @@
+/**
+ * Parses the value of "Value" and reflects it back as attribute.
+ */
+export async function handler(event: any) {
+  return {
+    Data: {
+      Value: JSON.parse(event.ResourceProperties.Value),
```

Comment:
I'm confused. Are you expecting the custom resource invocation to look like:

```
abcd: {
  Type: "Custom::AWSCDKCfnJson",
  Properties: {
    Value: "{ \"Ref\": \"Something\" }"
  }
}
```

?

Why not simply:

```
abcd: {
  Type: "Custom::AWSCDKCfnJson",
  Properties: {
    Value: { Ref: "Something" }
  }
}
```
?

Reply:
Bee tee dubs the consumer kinda seems to be expecting that this will turn out to be a string. Validation mayhaps?

Reply:
The main use case for this is to allow using intrinsics as *keys* in JSON hashes. If I don't serialize this to a string during synthesis, I am not solving this problem.

For example:

```json
"abcd": {
  "Type": "Custom::AWSCDKCfnJson",
  "Properties": {
    "Value": {
        { Ref: "Something" }: "Foo"
    }
  }
}
```